plugins/catalog-backend/src/ingestion/processors/BitbucketApiReaderProcessor.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+import fetch, { RequestInit, HeadersInit } from 'node-fetch';
+import * as result from './results';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+
+export class BitbucketApiReaderProcessor implements LocationProcessor {
+  private username: string = process.env.BITBUCKET_USERNAME || '';
+  private password: string = process.env.BITBUCKET_PASSWORD || '';
```

Comment:
Could we use personal access tokens instead? https://confluence.atlassian.com/bitbucketserver/personal-access-tokens-939515499.html

Reply:
Thanks for the review @Rugvip 
Bitbucket cloud (bitbucket.org) does not support personal access tokens, it supports App Passwords. https://community.atlassian.com/t5/Bitbucket-questions/Personal-Access-Token-for-Bitbucket-Cloud/qaq-p/677663

Reply:
Oh what the shit, alright then :grin:

Maybe we should make it `BITBUCKET_APP_PASSWORD` though, just to avoid someone putting their actual password there?

Reply:
Have updated it to `BITBUCKET_APP_PASSWORD`